### ISSUE TITLE
Problems with nested Proto

### ISSUE THREAD

**Issue by alexbiehl** [k-bx/protocol-buffers#10]
Hi, 

I have following problem: 

A.proto:

```
package component.a;

import "B.proto";

message Package {
  required PackageId packageId = 1;
}
```

B.proto:

```
package component;

message PackageId {
    required string id = 1;
}
```

A.proto lives in package `components.a` and includes B.proto, which lives in package `components`. When running `hprotoc A.proto`it fails with: 

```
hprotoc: user error (Parsing proto:
"A.proto"
has failed with message
Text.ProtocolBuffers.ProtoCompile.Resolve annErr: fqFileDP FileDescriptorProto (name,package) is (Just "A.proto",Just "component.a")
  Text.ProtocolBuffers.ProtoCompile.Resolve annErr: fqMessage DescriptorProto name is Just "Package"
    Text.ProtocolBuffers.ProtoCompile.Resolve annErr: fqField FieldDescriptorProto name is Just "packageId"
      Text.ProtocolBuffers.ProtoCompile.Resolve fatal error encountered, message:
        resolvePredEnv: Could not lookup "PackageId"
        which parses as (False,[IName "PackageId"])
        in environment: .component.a.Package in "A.proto"
        looking for: Message or Group or Enum
        allowed (local):  [([IName "component",IName "a",IName "Package"],[IName "name",IName "packageId"])]
        allowed (global): [(PackageID {_getPackageID = [IName "component",IName "a"]},[IName "Package"]),(PackageID {_getPackageID = [IName "component"]},[IName "PackageId"])]

)
```

As you can see, `PackageId` is in the global env. A quick look in `Resolver.hs` revealed that global env is only searched in when the name to search for is qualified and preceeded by a `.`. Since googles protoc supports this this should be fixed. 

A solution could be, if we cannot find a name in local env, try again with globalenv, regardles of qualified name and such. Though that could wind up in name clashes in other cases, I am not sure. What do you think?

**Comment by k-bx** [comment]
Fixed in 2.4.1

**Comment by k-bx** [comment]
there was a version-constraint bug, fixed with 2.4.2
